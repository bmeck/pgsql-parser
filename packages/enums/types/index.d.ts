import toInt from './enums/enums2int.json';
import toStr from './enums/enums2str.json';
import nodes from './nodes.json';
export { toInt, toStr, nodes };
export declare const getEnum: (enumType: any, key: any) => any;
export * from './types';
