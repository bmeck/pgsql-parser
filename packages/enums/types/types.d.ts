export namespace TYPE_NAMES {
    let OBJECT_ACCESS_METHOD: string;
    let OBJECT_AGGREGATE: string;
    let OBJECT_ATTRIBUTE: string;
    let OBJECT_CAST: string;
    let OBJECT_COLLATION: string;
    let OBJECT_COLUMN: string;
    let OBJECT_CONVERSION: string;
    let OBJECT_DATABASE: string;
    let OBJECT_DOMAIN: string;
    let OBJECT_DOMCONSTRAINT: string;
    let OBJECT_EVENT_TRIGGER: string;
    let OBJECT_EXTENSION: string;
    let OBJECT_FDW: string;
    let OBJECT_FOREIGN_SERVER: string;
    let OBJECT_FOREIGN_TABLE: string;
    let OBJECT_FUNCTION: string;
    let OBJECT_INDEX: string;
    let OBJECT_LANGUAGE: string;
    let OBJECT_LARGEOBJECT: string;
    let OBJECT_MATVIEW: string;
    let OBJECT_OPCLASS: string;
    let OBJECT_OPERATOR: string;
    let OBJECT_OPFAMILY: string;
    let OBJECT_POLICY: string;
    let OBJECT_PROCEDURE: string;
    let OBJECT_PUBLICATION: string;
    let OBJECT_ROLE: string;
    let OBJECT_ROUTINE: string;
    let OBJECT_RULE: string;
    let OBJECT_SCHEMA: string;
    let OBJECT_SEQUENCE: string;
    let OBJECT_STATISTIC_EXT: string;
    let OBJECT_SUBSCRIPTION: string;
    let OBJECT_TABCONSTRAINT: string;
    let OBJECT_TABLE: string;
    let OBJECT_TABLESPACE: string;
    let OBJECT_TRANSFORM: string;
    let OBJECT_TRIGGER: string;
    let OBJECT_TSCONFIGURATION: string;
    let OBJECT_TSDICTIONARY: string;
    let OBJECT_TSPARSER: string;
    let OBJECT_TSTEMPLATE: string;
    let OBJECT_TYPE: string;
    let OBJECT_VIEW: string;
}
export function objtypeName(arg: any): any;
export function getConstraintFromConstrType(type: any): "NULL" | "NOT NULL" | "DEFAULT" | "CHECK" | "PRIMARY KEY" | "UNIQUE" | "EXCLUDE" | "REFERENCES" | "GENERATED" | "IDENTITY";
