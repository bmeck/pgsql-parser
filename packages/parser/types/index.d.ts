import { Node } from '@pgsql/types';
import { parsePlPgSQLSync as parseFunction } from 'libpg-query';
export declare const parse: (sql: any) => Node;
export declare const parseAsync: (sql: any) => Promise<any>;
export { parseFunction };
