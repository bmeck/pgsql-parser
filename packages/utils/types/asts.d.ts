import { ParseResult, ScanResult, Integer, Float, String, BitString, Null, List, OidList, IntList, Alias, RangeVar, TableFunc, Expr, Var, Param, Aggref, GroupingFunc, WindowFunc, SubscriptingRef, FuncExpr, NamedArgExpr, OpExpr, DistinctExpr, NullIfExpr, ScalarArrayOpExpr, BoolExpr, SubLink, SubPlan, AlternativeSubPlan, FieldSelect, FieldStore, RelabelType, CoerceViaIO, ArrayCoerceExpr, ConvertRowtypeExpr, CollateExpr, CaseExpr, CaseWhen, CaseTestExpr, ArrayExpr, RowExpr, RowCompareExpr, CoalesceExpr, MinMaxExpr, SQLValueFunction, XmlExpr, NullTest, BooleanTest, CoerceToDomain, CoerceToDomainValue, SetToDefault, CurrentOfExpr, NextValueExpr, InferenceElem, TargetEntry, RangeTblRef, JoinExpr, FromExpr, OnConflictExpr, IntoClause, RawStmt, Query, InsertStmt, DeleteStmt, UpdateStmt, SelectStmt, AlterTableStmt, AlterTableCmd, AlterDomainStmt, SetOperationStmt, GrantStmt, GrantRoleStmt, AlterDefaultPrivilegesStmt, ClosePortalStmt, ClusterStmt, CopyStmt, CreateStmt, DefineStmt, DropStmt, TruncateStmt, CommentStmt, FetchStmt, IndexStmt, CreateFunctionStmt, AlterFunctionStmt, DoStmt, RenameStmt, RuleStmt, NotifyStmt, ListenStmt, UnlistenStmt, TransactionStmt, ViewStmt, LoadStmt, CreateDomainStmt, CreatedbStmt, DropdbStmt, VacuumStmt, ExplainStmt, CreateTableAsStmt, CreateSeqStmt, AlterSeqStmt, VariableSetStmt, VariableShowStmt, DiscardStmt, CreateTrigStmt, CreatePLangStmt, CreateRoleStmt, AlterRoleStmt, DropRoleStmt, LockStmt, ConstraintsSetStmt, ReindexStmt, CheckPointStmt, CreateSchemaStmt, AlterDatabaseStmt, AlterDatabaseSetStmt, AlterRoleSetStmt, CreateConversionStmt, CreateCastStmt, CreateOpClassStmt, CreateOpFamilyStmt, AlterOpFamilyStmt, PrepareStmt, ExecuteStmt, DeallocateStmt, DeclareCursorStmt, CreateTableSpaceStmt, DropTableSpaceStmt, AlterObjectDependsStmt, AlterObjectSchemaStmt, AlterOwnerStmt, AlterOperatorStmt, AlterTypeStmt, DropOwnedStmt, ReassignOwnedStmt, CompositeTypeStmt, CreateEnumStmt, CreateRangeStmt, AlterEnumStmt, AlterTSDictionaryStmt, AlterTSConfigurationStmt, CreateFdwStmt, AlterFdwStmt, CreateForeignServerStmt, AlterForeignServerStmt, CreateUserMappingStmt, AlterUserMappingStmt, DropUserMappingStmt, AlterTableSpaceOptionsStmt, AlterTableMoveAllStmt, SecLabelStmt, CreateForeignTableStmt, ImportForeignSchemaStmt, CreateExtensionStmt, AlterExtensionStmt, AlterExtensionContentsStmt, CreateEventTrigStmt, AlterEventTrigStmt, RefreshMatViewStmt, ReplicaIdentityStmt, AlterSystemStmt, CreatePolicyStmt, AlterPolicyStmt, CreateTransformStmt, CreateAmStmt, CreatePublicationStmt, AlterPublicationStmt, CreateSubscriptionStmt, AlterSubscriptionStmt, DropSubscriptionStmt, CreateStatsStmt, AlterCollationStmt, CallStmt, AlterStatsStmt, A_Expr, ColumnRef, ParamRef, A_Const, FuncCall, A_Star, A_Indices, A_Indirection, A_ArrayExpr, ResTarget, MultiAssignRef, TypeCast, CollateClause, SortBy, WindowDef, RangeSubselect, RangeFunction, RangeTableSample, RangeTableFunc, RangeTableFuncCol, TypeName, ColumnDef, IndexElem, Constraint, DefElem, RangeTblEntry, RangeTblFunction, TableSampleClause, WithCheckOption, SortGroupClause, GroupingSet, WindowClause, ObjectWithArgs, AccessPriv, CreateOpClassItem, TableLikeClause, FunctionParameter, LockingClause, RowMarkClause, XmlSerialize, WithClause, InferClause, OnConflictClause, CommonTableExpr, RoleSpec, TriggerTransition, PartitionElem, PartitionSpec, PartitionBoundSpec, PartitionRangeDatum, PartitionCmd, VacuumRelation, InlineCodeBlock, CallContext, ScanToken } from "./wrapped";
declare const _default: {
    parseResult(_p?: ParseResult["ParseResult"]): ParseResult;
    scanResult(_p?: ScanResult["ScanResult"]): ScanResult;
    integer(_p?: Integer["Integer"]): Integer;
    float(_p?: Float["Float"]): Float;
    string(_p?: String["String"]): String;
    bitString(_p?: BitString["BitString"]): BitString;
    null(_p?: Null["Null"]): Null;
    list(_p?: List["List"]): List;
    oidList(_p?: OidList["OidList"]): OidList;
    intList(_p?: IntList["IntList"]): IntList;
    alias(_p?: Alias["Alias"]): Alias;
    rangeVar(_p?: RangeVar["RangeVar"]): RangeVar;
    tableFunc(_p?: TableFunc["TableFunc"]): TableFunc;
    expr(_p?: Expr["Expr"]): Expr;
    var(_p?: Var["Var"]): Var;
    param(_p?: Param["Param"]): Param;
    aggref(_p?: Aggref["Aggref"]): Aggref;
    groupingFunc(_p?: GroupingFunc["GroupingFunc"]): GroupingFunc;
    windowFunc(_p?: WindowFunc["WindowFunc"]): WindowFunc;
    subscriptingRef(_p?: SubscriptingRef["SubscriptingRef"]): SubscriptingRef;
    funcExpr(_p?: FuncExpr["FuncExpr"]): FuncExpr;
    namedArgExpr(_p?: NamedArgExpr["NamedArgExpr"]): NamedArgExpr;
    opExpr(_p?: OpExpr["OpExpr"]): OpExpr;
    distinctExpr(_p?: DistinctExpr["DistinctExpr"]): DistinctExpr;
    nullIfExpr(_p?: NullIfExpr["NullIfExpr"]): NullIfExpr;
    scalarArrayOpExpr(_p?: ScalarArrayOpExpr["ScalarArrayOpExpr"]): ScalarArrayOpExpr;
    boolExpr(_p?: BoolExpr["BoolExpr"]): BoolExpr;
    subLink(_p?: SubLink["SubLink"]): SubLink;
    subPlan(_p?: SubPlan["SubPlan"]): SubPlan;
    alternativeSubPlan(_p?: AlternativeSubPlan["AlternativeSubPlan"]): AlternativeSubPlan;
    fieldSelect(_p?: FieldSelect["FieldSelect"]): FieldSelect;
    fieldStore(_p?: FieldStore["FieldStore"]): FieldStore;
    relabelType(_p?: RelabelType["RelabelType"]): RelabelType;
    coerceViaio(_p?: CoerceViaIO["CoerceViaIO"]): CoerceViaIO;
    arrayCoerceExpr(_p?: ArrayCoerceExpr["ArrayCoerceExpr"]): ArrayCoerceExpr;
    convertRowtypeExpr(_p?: ConvertRowtypeExpr["ConvertRowtypeExpr"]): ConvertRowtypeExpr;
    collateExpr(_p?: CollateExpr["CollateExpr"]): CollateExpr;
    caseExpr(_p?: CaseExpr["CaseExpr"]): CaseExpr;
    caseWhen(_p?: CaseWhen["CaseWhen"]): CaseWhen;
    caseTestExpr(_p?: CaseTestExpr["CaseTestExpr"]): CaseTestExpr;
    arrayExpr(_p?: ArrayExpr["ArrayExpr"]): ArrayExpr;
    rowExpr(_p?: RowExpr["RowExpr"]): RowExpr;
    rowCompareExpr(_p?: RowCompareExpr["RowCompareExpr"]): RowCompareExpr;
    coalesceExpr(_p?: CoalesceExpr["CoalesceExpr"]): CoalesceExpr;
    minMaxExpr(_p?: MinMaxExpr["MinMaxExpr"]): MinMaxExpr;
    sqlValueFunction(_p?: SQLValueFunction["SQLValueFunction"]): SQLValueFunction;
    xmlExpr(_p?: XmlExpr["XmlExpr"]): XmlExpr;
    nullTest(_p?: NullTest["NullTest"]): NullTest;
    booleanTest(_p?: BooleanTest["BooleanTest"]): BooleanTest;
    coerceToDomain(_p?: CoerceToDomain["CoerceToDomain"]): CoerceToDomain;
    coerceToDomainValue(_p?: CoerceToDomainValue["CoerceToDomainValue"]): CoerceToDomainValue;
    setToDefault(_p?: SetToDefault["SetToDefault"]): SetToDefault;
    currentOfExpr(_p?: CurrentOfExpr["CurrentOfExpr"]): CurrentOfExpr;
    nextValueExpr(_p?: NextValueExpr["NextValueExpr"]): NextValueExpr;
    inferenceElem(_p?: InferenceElem["InferenceElem"]): InferenceElem;
    targetEntry(_p?: TargetEntry["TargetEntry"]): TargetEntry;
    rangeTblRef(_p?: RangeTblRef["RangeTblRef"]): RangeTblRef;
    joinExpr(_p?: JoinExpr["JoinExpr"]): JoinExpr;
    fromExpr(_p?: FromExpr["FromExpr"]): FromExpr;
    onConflictExpr(_p?: OnConflictExpr["OnConflictExpr"]): OnConflictExpr;
    intoClause(_p?: IntoClause["IntoClause"]): IntoClause;
    rawStmt(_p?: RawStmt["RawStmt"]): RawStmt;
    query(_p?: Query["Query"]): Query;
    insertStmt(_p?: InsertStmt["InsertStmt"]): InsertStmt;
    deleteStmt(_p?: DeleteStmt["DeleteStmt"]): DeleteStmt;
    updateStmt(_p?: UpdateStmt["UpdateStmt"]): UpdateStmt;
    selectStmt(_p?: SelectStmt["SelectStmt"]): SelectStmt;
    alterTableStmt(_p?: AlterTableStmt["AlterTableStmt"]): AlterTableStmt;
    alterTableCmd(_p?: AlterTableCmd["AlterTableCmd"]): AlterTableCmd;
    alterDomainStmt(_p?: AlterDomainStmt["AlterDomainStmt"]): AlterDomainStmt;
    setOperationStmt(_p?: SetOperationStmt["SetOperationStmt"]): SetOperationStmt;
    grantStmt(_p?: GrantStmt["GrantStmt"]): GrantStmt;
    grantRoleStmt(_p?: GrantRoleStmt["GrantRoleStmt"]): GrantRoleStmt;
    alterDefaultPrivilegesStmt(_p?: AlterDefaultPrivilegesStmt["AlterDefaultPrivilegesStmt"]): AlterDefaultPrivilegesStmt;
    closePortalStmt(_p?: ClosePortalStmt["ClosePortalStmt"]): ClosePortalStmt;
    clusterStmt(_p?: ClusterStmt["ClusterStmt"]): ClusterStmt;
    copyStmt(_p?: CopyStmt["CopyStmt"]): CopyStmt;
    createStmt(_p?: CreateStmt["CreateStmt"]): CreateStmt;
    defineStmt(_p?: DefineStmt["DefineStmt"]): DefineStmt;
    dropStmt(_p?: DropStmt["DropStmt"]): DropStmt;
    truncateStmt(_p?: TruncateStmt["TruncateStmt"]): TruncateStmt;
    commentStmt(_p?: CommentStmt["CommentStmt"]): CommentStmt;
    fetchStmt(_p?: FetchStmt["FetchStmt"]): FetchStmt;
    indexStmt(_p?: IndexStmt["IndexStmt"]): IndexStmt;
    createFunctionStmt(_p?: CreateFunctionStmt["CreateFunctionStmt"]): CreateFunctionStmt;
    alterFunctionStmt(_p?: AlterFunctionStmt["AlterFunctionStmt"]): AlterFunctionStmt;
    doStmt(_p?: DoStmt["DoStmt"]): DoStmt;
    renameStmt(_p?: RenameStmt["RenameStmt"]): RenameStmt;
    ruleStmt(_p?: RuleStmt["RuleStmt"]): RuleStmt;
    notifyStmt(_p?: NotifyStmt["NotifyStmt"]): NotifyStmt;
    listenStmt(_p?: ListenStmt["ListenStmt"]): ListenStmt;
    unlistenStmt(_p?: UnlistenStmt["UnlistenStmt"]): UnlistenStmt;
    transactionStmt(_p?: TransactionStmt["TransactionStmt"]): TransactionStmt;
    viewStmt(_p?: ViewStmt["ViewStmt"]): ViewStmt;
    loadStmt(_p?: LoadStmt["LoadStmt"]): LoadStmt;
    createDomainStmt(_p?: CreateDomainStmt["CreateDomainStmt"]): CreateDomainStmt;
    createdbStmt(_p?: CreatedbStmt["CreatedbStmt"]): CreatedbStmt;
    dropdbStmt(_p?: DropdbStmt["DropdbStmt"]): DropdbStmt;
    vacuumStmt(_p?: VacuumStmt["VacuumStmt"]): VacuumStmt;
    explainStmt(_p?: ExplainStmt["ExplainStmt"]): ExplainStmt;
    createTableAsStmt(_p?: CreateTableAsStmt["CreateTableAsStmt"]): CreateTableAsStmt;
    createSeqStmt(_p?: CreateSeqStmt["CreateSeqStmt"]): CreateSeqStmt;
    alterSeqStmt(_p?: AlterSeqStmt["AlterSeqStmt"]): AlterSeqStmt;
    variableSetStmt(_p?: VariableSetStmt["VariableSetStmt"]): VariableSetStmt;
    variableShowStmt(_p?: VariableShowStmt["VariableShowStmt"]): VariableShowStmt;
    discardStmt(_p?: DiscardStmt["DiscardStmt"]): DiscardStmt;
    createTrigStmt(_p?: CreateTrigStmt["CreateTrigStmt"]): CreateTrigStmt;
    createpLangStmt(_p?: CreatePLangStmt["CreatePLangStmt"]): CreatePLangStmt;
    createRoleStmt(_p?: CreateRoleStmt["CreateRoleStmt"]): CreateRoleStmt;
    alterRoleStmt(_p?: AlterRoleStmt["AlterRoleStmt"]): AlterRoleStmt;
    dropRoleStmt(_p?: DropRoleStmt["DropRoleStmt"]): DropRoleStmt;
    lockStmt(_p?: LockStmt["LockStmt"]): LockStmt;
    constraintsSetStmt(_p?: ConstraintsSetStmt["ConstraintsSetStmt"]): ConstraintsSetStmt;
    reindexStmt(_p?: ReindexStmt["ReindexStmt"]): ReindexStmt;
    checkPointStmt(_p?: CheckPointStmt["CheckPointStmt"]): CheckPointStmt;
    createSchemaStmt(_p?: CreateSchemaStmt["CreateSchemaStmt"]): CreateSchemaStmt;
    alterDatabaseStmt(_p?: AlterDatabaseStmt["AlterDatabaseStmt"]): AlterDatabaseStmt;
    alterDatabaseSetStmt(_p?: AlterDatabaseSetStmt["AlterDatabaseSetStmt"]): AlterDatabaseSetStmt;
    alterRoleSetStmt(_p?: AlterRoleSetStmt["AlterRoleSetStmt"]): AlterRoleSetStmt;
    createConversionStmt(_p?: CreateConversionStmt["CreateConversionStmt"]): CreateConversionStmt;
    createCastStmt(_p?: CreateCastStmt["CreateCastStmt"]): CreateCastStmt;
    createOpClassStmt(_p?: CreateOpClassStmt["CreateOpClassStmt"]): CreateOpClassStmt;
    createOpFamilyStmt(_p?: CreateOpFamilyStmt["CreateOpFamilyStmt"]): CreateOpFamilyStmt;
    alterOpFamilyStmt(_p?: AlterOpFamilyStmt["AlterOpFamilyStmt"]): AlterOpFamilyStmt;
    prepareStmt(_p?: PrepareStmt["PrepareStmt"]): PrepareStmt;
    executeStmt(_p?: ExecuteStmt["ExecuteStmt"]): ExecuteStmt;
    deallocateStmt(_p?: DeallocateStmt["DeallocateStmt"]): DeallocateStmt;
    declareCursorStmt(_p?: DeclareCursorStmt["DeclareCursorStmt"]): DeclareCursorStmt;
    createTableSpaceStmt(_p?: CreateTableSpaceStmt["CreateTableSpaceStmt"]): CreateTableSpaceStmt;
    dropTableSpaceStmt(_p?: DropTableSpaceStmt["DropTableSpaceStmt"]): DropTableSpaceStmt;
    alterObjectDependsStmt(_p?: AlterObjectDependsStmt["AlterObjectDependsStmt"]): AlterObjectDependsStmt;
    alterObjectSchemaStmt(_p?: AlterObjectSchemaStmt["AlterObjectSchemaStmt"]): AlterObjectSchemaStmt;
    alterOwnerStmt(_p?: AlterOwnerStmt["AlterOwnerStmt"]): AlterOwnerStmt;
    alterOperatorStmt(_p?: AlterOperatorStmt["AlterOperatorStmt"]): AlterOperatorStmt;
    alterTypeStmt(_p?: AlterTypeStmt["AlterTypeStmt"]): AlterTypeStmt;
    dropOwnedStmt(_p?: DropOwnedStmt["DropOwnedStmt"]): DropOwnedStmt;
    reassignOwnedStmt(_p?: ReassignOwnedStmt["ReassignOwnedStmt"]): ReassignOwnedStmt;
    compositeTypeStmt(_p?: CompositeTypeStmt["CompositeTypeStmt"]): CompositeTypeStmt;
    createEnumStmt(_p?: CreateEnumStmt["CreateEnumStmt"]): CreateEnumStmt;
    createRangeStmt(_p?: CreateRangeStmt["CreateRangeStmt"]): CreateRangeStmt;
    alterEnumStmt(_p?: AlterEnumStmt["AlterEnumStmt"]): AlterEnumStmt;
    altertsDictionaryStmt(_p?: AlterTSDictionaryStmt["AlterTSDictionaryStmt"]): AlterTSDictionaryStmt;
    altertsConfigurationStmt(_p?: AlterTSConfigurationStmt["AlterTSConfigurationStmt"]): AlterTSConfigurationStmt;
    createFdwStmt(_p?: CreateFdwStmt["CreateFdwStmt"]): CreateFdwStmt;
    alterFdwStmt(_p?: AlterFdwStmt["AlterFdwStmt"]): AlterFdwStmt;
    createForeignServerStmt(_p?: CreateForeignServerStmt["CreateForeignServerStmt"]): CreateForeignServerStmt;
    alterForeignServerStmt(_p?: AlterForeignServerStmt["AlterForeignServerStmt"]): AlterForeignServerStmt;
    createUserMappingStmt(_p?: CreateUserMappingStmt["CreateUserMappingStmt"]): CreateUserMappingStmt;
    alterUserMappingStmt(_p?: AlterUserMappingStmt["AlterUserMappingStmt"]): AlterUserMappingStmt;
    dropUserMappingStmt(_p?: DropUserMappingStmt["DropUserMappingStmt"]): DropUserMappingStmt;
    alterTableSpaceOptionsStmt(_p?: AlterTableSpaceOptionsStmt["AlterTableSpaceOptionsStmt"]): AlterTableSpaceOptionsStmt;
    alterTableMoveAllStmt(_p?: AlterTableMoveAllStmt["AlterTableMoveAllStmt"]): AlterTableMoveAllStmt;
    secLabelStmt(_p?: SecLabelStmt["SecLabelStmt"]): SecLabelStmt;
    createForeignTableStmt(_p?: CreateForeignTableStmt["CreateForeignTableStmt"]): CreateForeignTableStmt;
    importForeignSchemaStmt(_p?: ImportForeignSchemaStmt["ImportForeignSchemaStmt"]): ImportForeignSchemaStmt;
    createExtensionStmt(_p?: CreateExtensionStmt["CreateExtensionStmt"]): CreateExtensionStmt;
    alterExtensionStmt(_p?: AlterExtensionStmt["AlterExtensionStmt"]): AlterExtensionStmt;
    alterExtensionContentsStmt(_p?: AlterExtensionContentsStmt["AlterExtensionContentsStmt"]): AlterExtensionContentsStmt;
    createEventTrigStmt(_p?: CreateEventTrigStmt["CreateEventTrigStmt"]): CreateEventTrigStmt;
    alterEventTrigStmt(_p?: AlterEventTrigStmt["AlterEventTrigStmt"]): AlterEventTrigStmt;
    refreshMatViewStmt(_p?: RefreshMatViewStmt["RefreshMatViewStmt"]): RefreshMatViewStmt;
    replicaIdentityStmt(_p?: ReplicaIdentityStmt["ReplicaIdentityStmt"]): ReplicaIdentityStmt;
    alterSystemStmt(_p?: AlterSystemStmt["AlterSystemStmt"]): AlterSystemStmt;
    createPolicyStmt(_p?: CreatePolicyStmt["CreatePolicyStmt"]): CreatePolicyStmt;
    alterPolicyStmt(_p?: AlterPolicyStmt["AlterPolicyStmt"]): AlterPolicyStmt;
    createTransformStmt(_p?: CreateTransformStmt["CreateTransformStmt"]): CreateTransformStmt;
    createAmStmt(_p?: CreateAmStmt["CreateAmStmt"]): CreateAmStmt;
    createPublicationStmt(_p?: CreatePublicationStmt["CreatePublicationStmt"]): CreatePublicationStmt;
    alterPublicationStmt(_p?: AlterPublicationStmt["AlterPublicationStmt"]): AlterPublicationStmt;
    createSubscriptionStmt(_p?: CreateSubscriptionStmt["CreateSubscriptionStmt"]): CreateSubscriptionStmt;
    alterSubscriptionStmt(_p?: AlterSubscriptionStmt["AlterSubscriptionStmt"]): AlterSubscriptionStmt;
    dropSubscriptionStmt(_p?: DropSubscriptionStmt["DropSubscriptionStmt"]): DropSubscriptionStmt;
    createStatsStmt(_p?: CreateStatsStmt["CreateStatsStmt"]): CreateStatsStmt;
    alterCollationStmt(_p?: AlterCollationStmt["AlterCollationStmt"]): AlterCollationStmt;
    callStmt(_p?: CallStmt["CallStmt"]): CallStmt;
    alterStatsStmt(_p?: AlterStatsStmt["AlterStatsStmt"]): AlterStatsStmt;
    aExpr(_p?: A_Expr["A_Expr"]): A_Expr;
    columnRef(_p?: ColumnRef["ColumnRef"]): ColumnRef;
    paramRef(_p?: ParamRef["ParamRef"]): ParamRef;
    aConst(_p?: A_Const["A_Const"]): A_Const;
    funcCall(_p?: FuncCall["FuncCall"]): FuncCall;
    aStar(_p?: A_Star["A_Star"]): A_Star;
    aIndices(_p?: A_Indices["A_Indices"]): A_Indices;
    aIndirection(_p?: A_Indirection["A_Indirection"]): A_Indirection;
    aArrayExpr(_p?: A_ArrayExpr["A_ArrayExpr"]): A_ArrayExpr;
    resTarget(_p?: ResTarget["ResTarget"]): ResTarget;
    multiAssignRef(_p?: MultiAssignRef["MultiAssignRef"]): MultiAssignRef;
    typeCast(_p?: TypeCast["TypeCast"]): TypeCast;
    collateClause(_p?: CollateClause["CollateClause"]): CollateClause;
    sortBy(_p?: SortBy["SortBy"]): SortBy;
    windowDef(_p?: WindowDef["WindowDef"]): WindowDef;
    rangeSubselect(_p?: RangeSubselect["RangeSubselect"]): RangeSubselect;
    rangeFunction(_p?: RangeFunction["RangeFunction"]): RangeFunction;
    rangeTableSample(_p?: RangeTableSample["RangeTableSample"]): RangeTableSample;
    rangeTableFunc(_p?: RangeTableFunc["RangeTableFunc"]): RangeTableFunc;
    rangeTableFuncCol(_p?: RangeTableFuncCol["RangeTableFuncCol"]): RangeTableFuncCol;
    typeName(_p?: TypeName): TypeName;
    columnDef(_p?: ColumnDef["ColumnDef"]): ColumnDef;
    indexElem(_p?: IndexElem["IndexElem"]): IndexElem;
    constraint(_p?: Constraint["Constraint"]): Constraint;
    defElem(_p?: DefElem["DefElem"]): DefElem;
    rangeTblEntry(_p?: RangeTblEntry["RangeTblEntry"]): RangeTblEntry;
    rangeTblFunction(_p?: RangeTblFunction["RangeTblFunction"]): RangeTblFunction;
    tableSampleClause(_p?: TableSampleClause["TableSampleClause"]): TableSampleClause;
    withCheckOption(_p?: WithCheckOption["WithCheckOption"]): WithCheckOption;
    sortGroupClause(_p?: SortGroupClause["SortGroupClause"]): SortGroupClause;
    groupingSet(_p?: GroupingSet["GroupingSet"]): GroupingSet;
    windowClause(_p?: WindowClause["WindowClause"]): WindowClause;
    objectWithArgs(_p?: ObjectWithArgs["ObjectWithArgs"]): ObjectWithArgs;
    accessPriv(_p?: AccessPriv["AccessPriv"]): AccessPriv;
    createOpClassItem(_p?: CreateOpClassItem["CreateOpClassItem"]): CreateOpClassItem;
    tableLikeClause(_p?: TableLikeClause["TableLikeClause"]): TableLikeClause;
    functionParameter(_p?: FunctionParameter["FunctionParameter"]): FunctionParameter;
    lockingClause(_p?: LockingClause["LockingClause"]): LockingClause;
    rowMarkClause(_p?: RowMarkClause["RowMarkClause"]): RowMarkClause;
    xmlSerialize(_p?: XmlSerialize["XmlSerialize"]): XmlSerialize;
    withClause(_p?: WithClause["WithClause"]): WithClause;
    inferClause(_p?: InferClause["InferClause"]): InferClause;
    onConflictClause(_p?: OnConflictClause["OnConflictClause"]): OnConflictClause;
    commonTableExpr(_p?: CommonTableExpr["CommonTableExpr"]): CommonTableExpr;
    roleSpec(_p?: RoleSpec["RoleSpec"]): RoleSpec;
    triggerTransition(_p?: TriggerTransition["TriggerTransition"]): TriggerTransition;
    partitionElem(_p?: PartitionElem["PartitionElem"]): PartitionElem;
    partitionSpec(_p?: PartitionSpec["PartitionSpec"]): PartitionSpec;
    partitionBoundSpec(_p?: PartitionBoundSpec["PartitionBoundSpec"]): PartitionBoundSpec;
    partitionRangeDatum(_p?: PartitionRangeDatum["PartitionRangeDatum"]): PartitionRangeDatum;
    partitionCmd(_p?: PartitionCmd["PartitionCmd"]): PartitionCmd;
    vacuumRelation(_p?: VacuumRelation["VacuumRelation"]): VacuumRelation;
    inlineCodeBlock(_p?: InlineCodeBlock["InlineCodeBlock"]): InlineCodeBlock;
    callContext(_p?: CallContext["CallContext"]): CallContext;
    scanToken(_p?: ScanToken["ScanToken"]): ScanToken;
};
export default _default;
