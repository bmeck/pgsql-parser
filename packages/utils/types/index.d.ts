export * from './wrapped';
export * from './utils';
import asts from './asts';
export default asts;
