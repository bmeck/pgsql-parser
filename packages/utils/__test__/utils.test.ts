import * as u from '../src';
import ast from '../src/asts';
import { CreateStmt, ColumnDef } from '@pgsql/types';

it('getEnumValue', () => {
  expect(u.getEnumValue('A_Expr_Kind', 0)).toMatchSnapshot();
  expect(u.getEnumValue('A_Expr_Kind', 1)).toMatchSnapshot();
  expect(u.getEnumValue('A_Expr_Kind', 2)).toMatchSnapshot();
  expect(u.getEnumValue('A_Expr_Kind', 3)).toMatchSnapshot();
  expect(u.getEnumValue('A_Expr_Kind', 4)).toMatchSnapshot();
  expect(u.getEnumValue('A_Expr_Kind', 5)).toMatchSnapshot();
});

it('asts', () => {
  const newColumn: ColumnDef = ast.columnDef({
    colname: 'id',
    typeName: ast.typeName({
      names: [ast.string({ str: 'int4' })]
    })
  });

  const createStmt: CreateStmt = ast.createStmt({
    relation: ast.rangeVar({
      relname: 'new_table'
    }),
    tableElts: [newColumn]
  })
  expect(createStmt).toBeTruthy();
});