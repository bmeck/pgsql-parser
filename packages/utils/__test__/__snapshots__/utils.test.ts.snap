// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dynamic creation of tables 1`] = `
"CREATE TABLE users (
 	id int PRIMARY KEY,
	username text,
	email text UNIQUE,
	created_at timestamp NOT NULL 
)"
`;

exports[`getEnumValue snapshots 1`] = `"AEXPR_OP"`;

exports[`getEnumValue snapshots 2`] = `"AEXPR_OP_ANY"`;

exports[`getEnumValue snapshots 3`] = `"AEXPR_OP_ALL"`;

exports[`getEnumValue snapshots 4`] = `"AEXPR_DISTINCT"`;

exports[`getEnumValue snapshots 5`] = `"AEXPR_NOT_DISTINCT"`;

exports[`getEnumValue snapshots 6`] = `"AEXPR_NULLIF"`;

exports[`simple SelectStmt 1`] = `
{
  "SelectStmt": {
    "fromClause": [
      {
        "RangeVar": {
          "inh": true,
          "relname": "another_table",
          "relpersistence": "p",
        },
      },
    ],
    "limitOption": "LIMIT_OPTION_DEFAULT",
    "op": "SETOP_NONE",
    "targetList": [
      {
        "ResTarget": {
          "val": {
            "ColumnRef": {
              "fields": [
                {
                  "A_Star": {},
                },
              ],
            },
          },
        },
      },
    ],
  },
}
`;

exports[`simple SelectStmt 2`] = `"SELECT * FROM another_table"`;
